Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/checkins/CheckInsViewModel.kt

```diff
@@ -68,6 +76,22 @@ class CheckInsViewModel @AssistedInject constructor(
         savedState.set(SKEY_LAST_DEEPLINK, deepLink)
     }
 
+    fun onRemoveCheckInConfirmed(checkIn: CheckIn?) {
+        Timber.d("removeCheckin(checkIn=%s)", checkIn)
+        launch(scope = appScope) {
```

Comment:
Because of the `withContext(NonCancellable)`, I think we don't need to use the `appScope` and can simply use the `viewModelScope`

Reply:
I think there might still be a racecondition.
Maybe, I'll think about this a bit more and fix it in a follow up PR if necessary.

```kotlin
    launch(scope = appScope) {
            if (checkIn == null) { // <------ If we cancel the scope here, is `clear` still called? 
                checkInsRepository.clear()
            } else {
                checkInsRepository.deleteCheckIns(listOf(checkIn))
            }
     }
```

Reply:
Oh, good point. I am pretty sure `clear()` wouldn't be called in the situation you described, because the suspend function `withContext()` would immediately throw a `CancellationException` and the "NonCancellable" code block wouldn't be executed. 

=> Leave it as it is :)